File: remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode slowPtr = head;
        ListNode fastPtr = head;
        
        while (n > 0) {
            fastPtr = fastPtr.next;
            n--;
        }
        
        if (fastPtr == null) {
            return head.next;
        }
        
        while (fastPtr.next != null) {
            fastPtr = fastPtr.next;
            slowPtr = slowPtr.next;
        }
        
        slowPtr.next = slowPtr.next.next;
        
        return head;
    }
}

File: remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        
        fast_ptr = head
        for _ in range(n):
            fast_ptr = fast_ptr.next
        
        slow_ptr = head
        
        # Remove head
        if fast_ptr is None:
            return head.next
        
        while fast_ptr.next is not None:
            slow_ptr = slow_ptr.next
            fast_ptr = fast_ptr.next
        
        
        slow_ptr.next = slow_ptr.next.next
        
        return head